tests/baselines/reference/arrayFilterBooleanExternalOverload4.types

```diff
@@ -49,37 +49,37 @@ const t11 = (0 as any as F2<1,1>)(id(),id());
 >id : ID
 

```

Comment:
The change in behavior is easiest to see in this test file.  The evaluation of `id()` got stuck in earlier overloads, resulting in incorrect overload choice behavior..

`id()` is used in multiple positions to shows that multiple nodes are being cleared for the same iteration.